deployments.go

```diff
@@ -58,25 +59,43 @@ func (s *deployerService) deploy(ctx context.Context, db *gorm.DB, opts DeployOp
 		Slug:        slug,
 		Description: desc,
 	})
-
 	return r, err
 }
 
+func (s *deployerService) deployInTransaction(ctx context.Context, stream scheduler.StatusStream, opts DeployOpts) (*Release, error) {
+	tx := s.db.Begin()
+	r, err := s.createRelease(ctx, tx, stream, opts)
+	if err != nil {
+		tx.Rollback()
+		return r, err
+	}
+	if err := tx.Commit().Error; err != nil {
+		return r, err
+	}
+	return r, s.releases.Release(ctx, r, stream)
+}
+
 // Deploy is a thin wrapper around deploy to that adds the error to the
 // jsonmessage stream.
-func (s *deployerService) Deploy(ctx context.Context, db *gorm.DB, opts DeployOpts) (*Release, error) {
+func (s *deployerService) Deploy(ctx context.Context, opts DeployOpts) (*Release, error) {
 	var msg jsonmessage.JSONMessage
 
-	r, err := s.deploy(ctx, db, opts)
+	var stream scheduler.StatusStream
+	if opts.Stream {
+		stream = scheduler.NewJSONMessageStream(opts.Output)
+	}
+
+	r, err := s.deployInTransaction(ctx, stream, opts)
 	if err != nil {
 		msg = newJSONMessageError(err)
-	} else {
+	} else if stream == nil {
 		msg = jsonmessage.JSONMessage{Status: fmt.Sprintf("Status: Created new release v%d for %s", r.Version, r.App.Name)}
+	} else {
+		scheduler.Publish(stream, fmt.Sprintf("Finished processing events for release v%d of %s", r.Version, r.App.Name))
```

Comment:
I think we should just set `msg =` like we do above. Right now, `msg` is nil, which results in an empty line getting added to the stream after this call:

```console
$ ./build/emp deploy remind101/acme-inc:master -s
master: Pulling from remind101/acme-inc
420890c9e918: Already exists
a3ed95caeb02: Already exists
1e14aab0083d: Already exists
4393485f2bb3: Already exists
a2a051cf14f7: Already exists
1f9177248efa: Already exists
Digest: sha256:d05682347b56191a950f55ab06f11a6c6e34041921af32281b01f4a6a9a0f75f
Status: Image is up to date for remind101/acme-inc:master
Status: Created cloudformation template: https://empire-77792028-templatebucket-195oucd149ybu.s3.amazonaws.com/acme-inc/72c3d272-2be3-49fa-a175-32daf22e6d57/9ef308a9614931cac961791097dbd97d5438280e
Status: Stack update submitted
Status: Waiting for stack update to complete
Status: Stack update complete
Status: Finished processing events for release v3 of acme-inc

$
```